### PR TITLE
Milestone: Add validation and test for validations

### PR DESCRIPTION
**Pull request by geekelo**
Add the following validations:
For the User model:

- [ ] Name must not be blank.

- [ ] PostsCounter must be an integer greater than or equal to zero.

For the Post model:

- [ ] Title must not be blank.

- [ ] Title must not exceed 250 characters.

- [ ] CommentsCounter must be an integer greater than or equal to zero.

- [ ] LikesCounter must be an integer greater than or equal to zero.

- [ ] Add unit specs for all of your models' methods and validations.

### CODE REVIEW

**Comment by topeogunleye** [comment]
@geekelo, could you please kindly fix your linter errors before I start reviewing .

**Comment by topeogunleye** [comment]
Hi @geekelo, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ⚠️

   - You have submitted a project that has failing linters, Kindly make sure your linters are all passing before submitting the PR.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

>Invalid Code Review Request **does not count** into the code reviews limit.

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  it { should validate_presence_of(:title) }
+  it { should validate_length_of(:title).is_at_most(250) }
+  it { should validate_numericality_of(:comments_counter).only_integer.is_greater_than_or_equal_to(0) }
+  it { should validate_numericality_of(:likes_counter).only_integer.is_greater_than_or_equal_to(0) }
+end
```
Comment (V-Blaze) [inline spec/models/user_spec.rb]: - You have done great so far :+1: But there are a few requirements you still need to meet before moving to the next milestone. :+1: :100: 

- [ ] You have done great adding validations to your models. but Looking at the screenshot below, you can see that your tests are failing. This is the test file for the `User` model but you seem to have the same tests you wrote for the Post model here. I'll suggest you actually set up your test suite correctly and write actual tests for the validation in the users model. ALso you are meant to write tests for all the methods in each model also. in this case, In the users model, you need to also test `three_most_recent_post` which is a method that's meant to be in the users model. 

here's an example and a guideline of what your `user_spec` could look like.
```ruby
RSpec.describe User, type: :model do
  subject { User.new(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.', posts_counter: 0) }

  before { subject.save }

  it 'name should be present' do
    subject.name = nil
    expect(subject).to_not be_valid
  end
```  

- - Failed tests screenshot reference:
 
Screenshot from 2023-10-07 20-01-22

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  it { should validate_presence_of(:title) }
+  it { should validate_length_of(:title).is_at_most(250) }
+  it { should validate_numericality_of(:comments_counter).only_integer.is_greater_than_or_equal_to(0) }
+  it { should validate_numericality_of(:likes_counter).only_integer.is_greater_than_or_equal_to(0) }
```
Comment (V-Blaze) [inline spec/models/post_spec.rb]: - [ ] Looking at the screenshot i added to the previous comment, you can see that these tests are failing also, I'll suggest you look in to these your assertions as they seem not to be written correctly. Also as stated in the project requirement, you are meant to test all the methods in the Post model along side the validations. I'll suggest you look into this. here's a suggestion of how a validadton test and a method test for this model should look like. :+1: :wave: :car: 

```ruby

RSpec.describe Post, type: :model do
  subject { Post.new(author: User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.', posts_counter: 0), title: 'Hello', text: 'This is my first post', likes_counter: 0, comments_counter: 0) }

  before { subject.save }

  it 'comments_counter must be a number' do
    subject.comments_counter = 'a'
    expect(subject).to_not be_valid
  end

  it 'five_most_recent_comments should return a right result' do
    subject.comments.create(text: 'This is my first comment')
    subject.comments.create(text: 'This is my second comment')
    subject.comments.create(text: 'This is my third comment')
    subject.comments.create(text: 'This is my fourth comment')
    subject.comments.create(text: 'This is my fifth comment')
    subject.comments.create(text: 'This is my sixth comment')
    subject.comments.create(text: 'This is my seventh comment')
    expect(subject.five_most_recent_comments.count).to eq(5)
  end
end
```

File: app/models/post.rb [hunk 3]
```diff
@@ -3,6 +3,10 @@ class Post < ApplicationRecord
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
 
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
```
Comment (V-Blaze) [inline app/models/post.rb]: > Add unit specs for all of your models' methods and validations.

- [ ] Looking at your model specs, you don't have any for `comment and like` model. According to the requirement, you are meant to `Add unit specs for all of your models' methods and validations.`. I'll suggest you add this for likes and comments too and remember to test all the validations and model methods. :+1: :waffle: :smile:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @geekelo 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Correct git flow✔
- Added validations correctly 👋 
-  Readme looks very professional ✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @geekelo, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**